docs/source/cuda_environment_variables.rst

```diff
@@ -0,0 +1,19 @@
+.. _cuda_environment_variables:
+
+CUDA Environment Variables
+==========================
+.. list-table::
+   :header-rows: 1
+
+   * - Variable
+     - Description
+   * - ``CUDA_VISIBLE_DEVICES``
+     - Comma-separated list of GPU device IDs that should be made available to CUDA runtime. If set to ``-1``, no GPUs are made available.
+   * - ``CUDA_LAUNCH_BLOCKING``
+     - If set to ``1``, makes CUDA calls synchronous. This can be useful for debugging.
+   * - ``TORCH_CUDNN_V8_API_LRU_CACHE_LIMIT``
+     - The cache limit for the cuDNN v8 API. This is used to limit the memory used by the cuDNN v8 API. The default value is 10000, which roughly corresponds to 2GiB assuming 200KiB per ExecutionPlan. Set to ``0`` for no limit or a negative value for no caching.
+   * - ``PYTORCH_NO_CUDA_MEMORY_CACHING=1``
+     - If set to ``1``, disables caching of memory allocations in CUDA. This can be useful for debugging.
+   * - ``PYTORCH_NVML_BASED_CUDA_CHECK=1``
+     - If set to ``1``, before importing PyTorch modules that check if CUDA is available, PyTorch will use NVML to check if the CUDA driver is functional instead of using the CUDA runtime. This can be helpful if forked processes fail with a CUDA initialization error.
```

Comment:
A few OTOH:
`TORCH_CUDNN_V8_API_DISABLED=1` (disables the cuDNN V8 API and falls back to v7)
`CUBLAS_WORKSPACE_CONFIG=:4096:8` (sets the workspace size available to cuBLAS)
`CUDNN_CONV_WSCAP_DBG` (similar to the above for cuBLAS but a limit for cuDNN workspace size)
`TORCH_ALLOW_TF32_CUBLAS_OVERRIDE=1` (force TF32 enablement, overrides `set_float32_matmul_precision` setting)
`NVIDIA_TF32_OVERRIDE=0` (disable TF32 globally across all kernels, overriding all PyTorch settings)
`TORCH_NCCL_USE_COMM_NONBLOCKING` (use nonblocking error handling in NCCL)
`TORCH_NCCL_AVOID_RECORD_STREAMS=0` (fall back to record streams-based synchronization behavior in NCCL)

Reply:
`CUBLASLT_WORKSPACE_SIZE` (similar to the cuBLAS env var but for cuBLASLt workspaces managed by PyTorch)

Reply:
Thank you! these are great additions, will add

Reply:
`TORCH_CUDNN_V8_API_DEBUG=1` (sanity check whether cuDNN V8 is being used, also shows heuristic mode IIRC)
`CUDNN_ERRATA_JSON_FILE=` (path to errata filter that can be passed to cuDNN to avoid specific engine configs, used primarily for debugging or to hardcode autotuning), https://github.com/NVIDIA/cudnn-frontend/blob/a86ad708db725e4d29919bb6fadf8e6cdfa5dc06/README.FE.0.x.md?plain=1#L26